crates/anemo/src/network/connection_manager.rs

```diff
@@ -743,6 +743,19 @@ impl KnownPeers {
         self.inner_mut().insert(peer_info.peer_id, peer_info)
     }
 
+    pub fn batch_update<'a>(
+        &'a self,
+        to_remove: impl Iterator<Item = &'a PeerId>,
```

Comment:
I think these lifetimes can be elided, but lets see what clippy says

Reply:
Lifetime is required for `Item = &'a PeerId`, otherwise rustc complains with error[E0658]: anonymous lifetimes in `impl Trait` are unstable. `&self` doesn't require lifetime though.